src/Shared/Model/ArtifactUri.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Model;
+
+using System;
+using System.IO;
+
+public struct ArtifactUri
```

Comment:
Nit: Do you think this could be a [`readonly record struct`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record)?